client/src/app/components/common/save-as-modal.component.ts

```diff
@@ -43,15 +43,25 @@ import {ModalService} from "../modal";
 `
 })
 export class SaveAsModalComponent {
+
+    /** Path of the source file that should be copied */
+    public filePath: string;
+
+    /** Toggles the cog loader */
     private isCreatingFile: boolean;
+
+    /** Holds the error object that should be shown below the form if exists */
     private error: {[message: string]: string};
+
+    /** New file destination entry form */
     private newFileForm: ControlGroup;
 
     constructor(private formBuilder: FormBuilder,
                 private eventHub: EventHubService,
-                private workspace: WorkspaceService,
                 private modal: ModalService) {
 
+        this.filePath = undefined;
```

Comment:
The design is good, but do you have any documentation on Typescript not setting values to undefined? I could not find anything on that, nor was I able to get that behaviour: https://jsbin.com/terivus/edit?html,js,console